feat: adds reference sketeton (#57)
GreptimeTeam/docs: docs/reference/configuration.md

# Configuration

This page describes methods for configuring GreptimeDB server settings. Configuration can be set in TOML file.

When a parameter is absent in the configuration file, then the default value is used.

> All sample configuration files are in project's [config](https://github.com/GreptimeTeam/greptimedb/tree/develop/config) folder.

Reviewer: ```suggestion
All sample configuration files are in the project's [config](https://github.com/GreptimeTeam/greptimedb/tree/develop/config) folder.
```

## Specify configuration file

> You can specify configuration file by command line arg `-c [file_path]`, for eaxmple:

Reviewer: ```suggestion
You can specify the configuration file by using the command line arg `-c [file_path]`, for example:
```

```sh
greptime standalone start -c config/standalone.example.toml
```

## Common configurations

Common configurations in `frontend` and `standalone` sub command:

```toml
[grpc_options]
addr = '0.0.0.0:4001'
runtime_size = 8

[mysql_options]
addr = '0.0.0.0:4002'
runtime_size = 2

[influxdb_options]
enable = true

[opentsdb_options]
addr = "0.0.0.0:4242"
enable = true
runtime_size = 2

[prometheus_options]
enable = true

[postgres_options]
addr = '0.0.0.0:4003'
runtime_size = 2
check_pwd = false
```

All these options are optional, the default values are list as above. If you want to disable some options such as disabling OpenTSDB protocol support, you can just remove the `prometheus_options` or set it's `enable` value to be `false`.

## Standalone

> When you start the GreptimeDB in standalone mode, you can configure it as below:

Reviewer: ```suggestion
When you use GreptimeDB in the standalone mode, you can configure it as below:
```

| opentsdb_options   |              |         | OpenTSDB Protocol options                                                       |
|                    | enable       | Boolean | Whether to enable OpenTSDB protocol in HTTP API, true by default                |
|                    | addr         | String  | OpenTSDB telnet API server address, "0.0.0.0:4242" by default                   |
|                    | runtime_size | Integer | The number of server worker threads, 2 by default                               |
| prometheus_options |              |         | Prometheus protocol options                                                     |
|                    | enable       | Boolean | Whether to enable Prometheus remote write and read in HTTP API, true by default |
| postgres_options   |              |         | PostgresSQL server options                                                      |
|                    | addr         | String  | Server address, '0.0.0.0:4003' by default                                       |
|                    | runtime_size | Integer | The number of server worker threads, 2 by default                               |

> |                    | check_pwd    | boolean | Wheter to check password, it's not supported right now, always false.           |

Reviewer 2: typo. Wheter should be whether

Author: Good catch!

The `meta_client_opts` configure the metasrv client confugrations, incuding:

* `metasrv_addr`, metasrv address.
* `timeout_millis`, operation timeout in milliseconds, 3000 by default.
* `connect_timeout_millis`, connect server timeout in milliseconds,5000 by default.

## Datanode in distributed mode

Configure datanode in distributed mode:

```toml
node_id = 42
mode = "distributed"
rpc_addr = '127.0.0.1:3001'
wal_dir = '/tmp/greptimedb/wal'
rpc_runtime_size = 8
mysql_addr = '127.0.0.1:3306'
mysql_runtime_size = 4

[storage]
type = 'File'
data_dir = '/tmp/greptimedb/data/'

[meta_client_opts]
metasrv_addr = "1.1.1.1:3002"
timeout_millis = 3000
connect_timeout_millis = 5000
tcp_nodelay = false
```

Datanode in distributed mode should set different `node_id` in different nodes.

## Metasrv configuration

A sample configurations:

```toml
bind_addr = '127.0.0.1:3002'
server_addr = '127.0.0.1:3002'
store_addr = '127.0.0.1:2379'
datanode_lease_secs = 30
```

| Key                 | Type    | Description                                                                                                             |   |

> |---------------------|---------|-------------------------------------------------------------------------------------------------------------------------|---|

Reviewer 3: ```suggestion
|---------------------|---------|-------------------------------------------------------------------------------------------------------------------------|
```

Looks like there is an empty column in the end. Other rows should be change correspondingly.

| bind_addr           | String  | The bind address of metasrv, '127.0.0.1:3002' by default.                                                                 |   |
| server_addr         | String  | The communication server address for frontend and datanode to connect metasrv,  '127.0.0.1:3002' by default for localhost |   |

> | store_addr          | String  | Etcd server address, '127.0.0.1:2379' by default                                                                        |   |

Reviewer 3: ```suggestion
| store_addr          | String  | etcd server address, '127.0.0.1:2379' by default                                                                        |   |
```
